### PR TITLE
Add "print types" to bracketed symbols

### LINKED ISSUE #270: [Feature request] Add way to interpolate decimal and binary numbers
At the moment, you can use `{symbol}` to interpolate in hexadecimal literals... but not decimal or binary ones, which can be an issue.

My proposal for implementing this in a way that's comfortable to work with is this: expand `{}` to work with expressions (as opposed to just symbols) within the brackets, and add 3 new functions:

* `DEC(number)` (or `DECSTR(number)`): convert a number to a decimal literal. `DEC(12)` ⇒ `"12"`
* `HEX(number)` (or `HEXSTR(number)`): convert a number to a hexadecimal literal. `HEX(12)` ⇒ `"$C"`
* `BIN(number)` (or `BINSTR(number)`): convert a number to a binary literal. `BIN(12)` ⇒ `"%1100"`

You'd then be able to use, for example, `{DEC(symbol)}` to interpolate in non-hexadecimal numbers.

If adding expression support to `{}` is too much, special syntax could instead be introduced (although that's much less versatile), but whatever the solution ends up being it's not quite right that hexadecimal interpolation is available while decimal and binary isn't.

This'd allow for more versatile interpolation into strings and the like.

### PR DESCRIPTION
**Pull request by ISSOtm**
Should partially cover #178 and close #270.
This allows printing numbers in different bases and without the dollar prefix.
This is especially useful in macros because the dollar isn't a valid character for symbol names, [requiring](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L119) [heavy](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L238) [`STRSUB`](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L247) [usage](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L263) [to get rid](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L281) [of it](https://github.com/ISSOtm/rgbds-structs/blob/39b4cd67fd0a1c843d011c41cecd09ece23561a9/structs.asm#L380).

The PR currently doesn't have the man pages generated due to #389.

### CODE REVIEW

**Comment by ISSOtm** [comment]
Fixed the following:
- `docs/rgbds.7.html` made empty due to a failed `make wwwman`
- Coding style not respected in `src/asm/asmy.y`
- Commit message would mistakenly close #178 when I'm not sure it should be auto-closed

File: src/asm/asmy.y [hunk 1]
```diff
@@ -94,8 +98,25 @@ size_t symvaluetostring(char *dest, size_t maxLength, char *sym)
...
+			char binary[32 + 1];
+			char *write_ptr = binary + 32;
+			fullLength = 0;
+			*write_ptr = 0;
+			do {
+				*(--write_ptr) = (value & 1) + '0';
+				value >>= 1;
+				fullLength++;
+			} while(value);
+			strncpy(dest, write_ptr, maxLength + 1);
```
Comment (NieDzejkob) [inline src/asm/asmy.y]: I think you have an unterminated string in `binary` here.

**Comment by NieDzejkob** [comment]
Personally, I wouldn't merge this without an addition to the test suite.

File: src/asm/asmy.y (hunk 1 above)
Comment (ISSOtm) [reply src/asm/asmy.y]: Except not because the string terminator is written on line 108

Comment (NieDzejkob) [reply src/asm/asmy.y]: Ah, right.

**Comment by ISSOtm** [comment]
I agree, but what should be tested?

**Comment by ISSOtm** [comment]
Fixed the following:
- `*write_ptr = 0;` was deemed too obtuse, replaced with `binary[32] = 0;`
- `Clarified `binary`'s array size

**Comment by NieDzejkob** [comment]
> I agree, but what should be tested?

Put numbers into a string with the new functionality. Use `WARN` to output them to stderr. Make sure the stderr looks as expected.

**Comment by ISSOtm** [comment]
I added some more test cases, that should be enough I think?

**Comment by NieDzejkob** [comment]
I just noticed the difference between `yyerror` and `fatalerror`. When the former triggers, no output file is generated and the exit code is 1, right?

**Comment by ISSOtm** [comment]
`yyerror` causes an error to occur, but compilation carries on. `fatalerror` immediately terminates compilation.

**Comment by ISSOtm** [comment]
(Bumping because CI failed on `./.travis-deps.sh`)

**Pull request merged**